lib/_http_common.js

```diff
@@ -236,48 +236,52 @@ exports.httpSocketSetup = httpSocketSetup;
  * using a regular expression since the former is up to 180% faster with v8 4.9
  * depending on the string length (the shorter the string, the larger the
  * performance difference)
+ *
+ * Additionally, checkIsHttpToken() is currently designed to be inlinable by v8,
+ * so take care when making changes to the implementation so that the source
+ * code size does not exceed v8's default max_inlined_source_size setting. Also,
+ * the character checking logic is moved to a separate inlinable function to
+ * allow some loop unrolling in checkIsHttpToken() while still remaining
+ * inlinable.
  **/
+function isValidTokenChar(ch) {
+  if (ch >= 94 && ch <= 122) // ^ _ ` a-z
+    return true;
+  if (ch >= 65 && ch <= 90) // A-Z
+    return true;
+  if (ch === 45) // -
+    return true;
+  if (ch >= 48 && ch <= 57) // 0-9
+    return true;
+  if (ch === 34 || ch === 40 || ch === 41 || ch === 44) // " ( ) ,
+    return false;
+  if (ch >= 33 && ch <= 46)
+    return true;
+  if (ch === 124 || ch === 126)
```

Comment:
Ok, now that this is split in two methods, are there still reasons to omit the comment here? This is ` // | ~`